### ISSUE TITLE
azurerm_application_gateway requires too many blocks

### ISSUE THREAD

**Issue by davidnortonjr** [hashicorp/terraform-provider-azurerm#15614]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

```
Terraform v1.1.6
on darwin_arm64
+ provider registry.terraform.io/hashicorp/azurerm v2.98.0
```

### Affected Resource(s)

* `azurerm_application_gateway`

### Terraform Configuration Files

```hcl
resource "azurerm_application_gateway" "mygateway" {
  name                = "mygateway"
  location            = "East US"
  resource_group_name = var.resource_group
  sku {
    name     = "Standard_Small"
    tier     = "Standard"
    capacity = 2
  }
}
```

### Expected Behaviour

I would expect to be able to create an application gateway without backends so I can [pass the ID into the Application Gateway Ingress Controller](https://docs.microsoft.com/en-us/azure/application-gateway/tutorial-ingress-controller-add-on-existing).

### Actual Behaviour

It requires a number of fields including:

- `backend_http_settings`
- `backend_address_pool`

### Steps to Reproduce

1. `terraform plan`

### Important Factoids

Just planning on creating this gateway in Terraform so I can point DNS at it, and then allow the AGIC in kubernetes to configure backends, frontends, etc.

using the command line, I am able to create a gateway without configuring backends: `az network application-gateway create --name mygateway --resource-group $resourceGroup`

**Comment by davidnortonjr** [comment]
Well, I looked deeper, and it appears that when you run the above azure CLI command, it actually creates something like. So perhaps the problem is more that the CLI picks a bunch of defaults, or that Azure requires these fields in the first place.

```
resource "azurerm_application_gateway" "mygateway" {
  name                = "mygateway"
  location            = "East US"
  resource_group_name = var.resource_group
  sku {
    name     = "Standard_Medium"
    tier     = "Standard"
    capacity = 2
  }
  backend_address_pool {
    name = "appGatewayBackendPool"
  }
  backend_http_settings {
    cookie_based_affinity = "Disabled"
    name                  = "appGatewayBackendHttpSettings"
    port                  = 80
    protocol              = "http"
    request_timeout       = 30
    connection_draining {
      drain_timeout_sec = 1
      enabled           = false
    }
  }
  frontend_ip_configuration {
    name = "appGatewayPrivateFrontendIP"
  }
  frontend_port {
    name = "appGatewayFrontendPort"
    port = 80
  }
  gateway_ip_configuration {
    name      = "appGatewayPrivateFrontendIP"
    subnet_id = var.subnet_id
  }
  http_listener {
    frontend_ip_configuration_name = "appGatewayPrivateFrontendIP"
    frontend_port_name             = "appGatewayFrontendPort"
    name                           = "appGatewayHttpListener"
    protocol                       = "http"
  }
  request_routing_rule {
    backend_address_pool_name  = "appGatewayBackendPool"
    backend_http_settings_name = "appGatewayBackendHttpSettings"
    http_listener_name         = "appGatewayHttpListener"
    name                       = "rule1"
    priority                   = 1
    rule_type                  = "Basic"
  }
}
```

**Comment by tombuildsstuff** [comment]
hi @davidnortonjr 

Thanks for opening this issue.

When provisioning an Application Gateway unfortunately it's required to create at least one of most of the sub-components, as such it's not possible to add support for these as separate resources without introducing a perectual diff on the main `azurerm_application_gateway` resource. This has come up a number of times over the years and is unfortunately blocked on the Azure Service Team who need to add support for this - which is currently being tracked [in this upstream issue](https://github.com/Azure/azure-rest-api-specs/issues/2313).

Unfortunately the Azure Networking team have since tried to move this Feature Request to the Azure Feedback site, which has since been shut down - as such [we've previously asked for an update](https://github.com/Azure/azure-rest-api-specs/issues/2313#issuecomment-997844834) to get this re-opened - however it may also be worth opening a new issue on that repository too.

Since the API doesn't allow provisioning an Application Gateway without any subcomponents at the moment unfortunately and so is a feature request for the Azure Service Team - I'm going to close this issue for the moment, but I'd recommend either commenting on the existing issue above, or making a new one so that the Service Team can take a look - but unfortunately there's not much we can do from Terraform's side at the moment.

Thanks!

**Comment by davidnortonjr** [comment]
@tombuildsstuff thank you very much for the detail response and the links! I will follow up there.
